random_joke.py

```diff
@@ -0,0 +1,30 @@
+#Flask app
+from flask import Flask
+import requests
+import json
+
+
+
+app = Flask(__name__)
+
+url = 'https://official-joke-api.appspot.com/random_joke'
+
+
+
+@app.route("/")
+def hello_world():
+    return "<h1>My first flask app!</h1>"
+
+@app.route("/random-joke")
+def return_joke():
+    response = requests.get(url)
+
+    #return api data as dictionary
+    api_dict = json.loads(response.text)
+
+    return f"{api_dict['setup']}...{api_dict['punchline']}"
```

Comment:
Great stuff! 🙌 

Often it is good practice to separate any code that calls an api into is own function or class then import it into wherever it needs to be used. This means that any other part of the code that needs to communicate with that api can use the same functionality. This is known as correctly separating concerns.

Such a class might look like:

```
from urllib.parse import urljoin

import requests

class JokeApi:
     
     base_url = https://official-joke-api.appspot.com
     
     @classmethod
     def get_random_joke(cls):
           endpoint = "random_joke"
           url = urljoin(cls.base_url, endpoint)
           response = requests.get(url)
           return response.json()
```

Then the `JokeApi` class can be imported an use in the `return_joke` function

Reply:
I Will read more about separating concerns.